core/src/main/scala/cats/data/package.scala

```diff
@@ -1,33 +1,14 @@
 package cats
 
 package object data {
-  type NonEmptyList[A] = OneAnd[List, A]
   type NonEmptyStream[A] = OneAnd[Stream, A]
   type ValidatedNel[E, A] = Validated[NonEmptyList[E], A]
 
-  def NonEmptyList[A](head: A, tail: List[A] = Nil): NonEmptyList[A] =
-    OneAnd(head, tail)
-  def NonEmptyList[A](head: A, tail: A*): NonEmptyList[A] =
-    OneAnd[List, A](head, tail.toList)
-
   def NonEmptyStream[A](head: A, tail: Stream[A] = Stream.empty): NonEmptyStream[A] =
     OneAnd(head, tail)
   def NonEmptyStream[A](head: A, tail: A*): NonEmptyStream[A] =
     OneAnd(head, tail.toStream)
 
-  object NonEmptyList {
-    def fromReducible[F[_], A](fa: F[A])(implicit F: Reducible[F]): Eval[NonEmptyList[A]] =
```

Comment:
I tend to somewhat agree on this. I'm not sure it is worth putting `to(|NonEmpty)(Vector|List|Stream)` in `Foldable|Reducible`.

I would be somewhat more inclined to put them on the companion objects of the data structure.

The general idea is something like:
```scala
trait CollectionyThingy[C[_]] {
  def singleton[T](t: T): C[T]
  // maybe want some Eval below
  def cons[T](head: T, tail: C[T]): C[T]
}

trait MaybeEmptyCollection[C[_]] extends CollectionyThingy[C] {
  def empty[T]: C[T]
}
```
but so far that seems lawless except `cons(a, empty) == singleton(a)`